proposals/ecosystem.bs

```diff
@@ -0,0 +1,1254 @@
+    * Ability to create new Notebooks that NeverNote will have permission to manage
+1. Alice later expands NeverNote’s access to include her Contacts
+1. Alice and Bob collaborate on a Notebook in Alice’s Pod. Bob is using a
+    different application than NeverNote called OtherNote.
+
+Application Registration {#appreg}
+=====================
+
+Issue: Consider incorporating a little bit of text here on *why* it is
+important for an application to be registered.
+
+## Checking registration status ## {#reg-status}
+
+When Alice arrives at NeverNote (`https://nevernote.example`) in her
+web browser, she is prompted to login with her [=WebID=] because NeverNote
+doesn’t know who she is yet. She could be a completely new user of
+NeverNote, or a returning user that has already registered NeverNote
+and authorized it to access her Notebooks and Notes.
+
+Alice logs into her [=identity provider=] (`https://idp.example`) with her
+WebID (`https://alice.pod.example/id#me`) and ultimately ends up with
+a [=token=] that NeverNote can use to make requests to her Pod.
+The token identifies her as Alice (`https://alice.pod.example/id#me`),
+using NeverNote (`https://nevernote.example/id#agent`) as a
+[=user-piloted application=].
+
+First NeverNote needs to check Alice’s [=application registries=] to see if
+there is already an [=application registration=] for NeverNote in one of
+them. If there is, than Nevernote is already registered, and doesn't
+need to go through the process again. If there is not, NeverNote will
+engage Alice's trusted [=application registration agent=] to register itself.
+
+### Discovering application registries ### {#reg-discovery}
+
+To discover Alice’s application registries, NeverNote must first dereference
+Alice's WebID to access [[#alice-profile]]. In that
+[=identity profile document=], NeverNote will look for an application
+[=registry set=]. There are registry sets of varying types listed
+in [[#alice-profile]]. Each registry set of a given type lists one or more
+registries of that type in different locations.
+
+<b>Excerpt from [[#alice-profile]]</b>
+<pre highlight="turtle">
+  <#me>
+    solid-ecosystem:applicationRegistrySet &lt;https://alice.pod.example/registries/application#set&gt; .
+</pre>
+
+Alice’s application registry set document is separate from her public
+identity profile document.
+This allows it to have a separate set of permissions.
+Registry set documents are not publicly accessible. They can only be
+accessed by Alice, or by a [=trusted agent=] where applicable.
+
+In [[#alice-appregset]] document, NeverNote sees that Alice has an
+application registry that stores application registrations in
+`<https://alice.pod.example/applications/>`.
+
+<b>Excerpt from [[#alice-appregset]] Document</b>
+<pre highlight="turtle">
+@prefix solid-ecosystem: &lt;http://www.w3.org/ns/solid/ecosystem#&gt; .
+
+<#set>
+  a solid-ecosystem:registrySet ;
+  solid-ecosystem:registryType solid-ecosystem:ApplicationRegistry ;
```

Comment:
I believe we'd want :ApplicationRegistry to be :applicationRegistry -- we seem to be inconsistent in the capitalization.

Reply:
`ApplicationRegistry` seems to be an instance; they are usually capitalized.

Reply:
Reading this made me wonder why they are not RDF types though.

Reply:
We had some debate on this -- landed on the case that (as of now) there is no difference between an application registry and a data registry, aside from the value of the :registryType predicate.

This :registryType is validated using the ShEx [here](https://github.com/solid/data-interoperability-panel/pull/42/files#diff-a2a3ce25ca75fa5d1aa87cbaa8c2602dR1034) to ensure a registry set for applications (ApplicationRegistry) does not have members of type DataRegistry.

Reply:
> Reading this made me wonder why they are not RDF types though.

RDF types would seem to make sense here. E.g.:

```turtle
ecosystem:Registry a rdfs:Class .
ecosystem:ApplicationRegistry a ecosystem:Registry .

<#set> a ecosystem:ApplicationRegistry .
```

Reply:
Fully expect and welcome tons of feedback on the data model. We put more of a premium on getting something together and presented soon even if it wasn't pristine yet so that we can all iterate on it together. We're reusing most of these patterns so the faster we can nail them down the cleaner the next submissions will be 🙂

Reply:
Created [this issue](https://github.com/solid/data-interoperability-panel/issues/45) to continue this conversation. We're going to merge and iterate on this through subsequent pulls and issues.